SpatialGDK/Documentation/content/porting-unreal-project-to-gdk.md

```diff
@@ -113,63 +117,69 @@ etc...
+    > If you ever need to run with default Unreal networking for workflow or validation reasons, untick `Spatial Networking` in the `Play` drop down menu in the Unreal editor. This flag is maintained in your Unreal config files under `/Script/EngineSettings.GeneralProjectSettings` and is also valid for non-editor builds.
     >
     > You can easily switch between networking modes using this checkbox. As the SpatialOS Unreal GDK is in pre-alpha, switching back to Unreal's networking may be a viable way to work around some issues you may encounter, and may speed up your iteration. However, in Unreal networking mode you lose access to key multi-server features of the GDK, which may lead to erratic behavior.
     >
     > In the future, we expect to achieve parity with Unreal networking in terms of iteration speed and stability, thus removing the need for a networking mode switch.
-1. Open the game's uproject file in the Editor and navigate to *Edit > Project Settings... > SpatialOS Unreal GDK > Toolbar*.
 
-1. Tag all replicated classes with the UCLASS tag `SpatialType`:
-    ``` 
-    UCLASS(SpatialType) 
-    ```
-1. Locate your game's `GameState`, and make it a public singleton:
+2. To provide context to SpatialOS on how to replicate singleton classes (such as GameState and GameMode) we have added a `UCLASS` macro which provides said context. The next steps detail how to do this. 
+   
+3. If your games '`GameState`' is a cpp file, locate it and mark it as a public singleton by modifying the UCLASS macro as shown:
     ```
     UCLASS(SpatialType=Singleton)
     ```
-1. Locate your game's `GameMode`, and make it a private singleton:
+    If your games '`GameState`' is a blueprint then you will need to open the full blueprint editor for that GameState and navigate to the `Class Settings` in the blueprint editor toolbar. In the `Class Options` tick the `Spatial Type` checkbox and add the tag `Singleton` to the `Spatial Description` textbox.
+
+4. If your games '`GameMode`' is a cpp file, locate it and mark it as a private singleton by modifying the UCLASS macro as shown:
     ```
     UCLASS(SpatialType=(Singleton,ServerOnly))
     ```
+    If your games '`GameMode`' is a blueprint then you will need to open the full blueprint editor for that GameMode and navigate to the `Class Settings` in the blueprint editor toolbar. In the `Class Options` tick the `Spatial Type` checkbox and add the tag `Singleton,ServerOnly` to the `Spatial Description` textbox.
+
+5. Marking these singleton classes as `Spatial Type` will cause spatial schema to be generated for said classes when pressing the `Schema` button in the `SpatialOSGDKToolbar`. We now need to mark the generated schema components as streaming queries for your UnrealWorker worker configuration, this ensures your UnrealWorkers always have these singletons checked out. The generated schema is located at `<ProjectRoot>\spatial\schema\improbable\unreal\generated`.
```

Comment:
Is it worth mentioning that this fiddly step will become obsolete in the near future?

Reply:
We'll update the docs when it's done I believe.